src/Decorator/DecoratorManager.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace src\Decorator;
+
+use DateTime;
+use Exception;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Log\LoggerInterface;
+use src\Integration\DataProvider;
+
+class DecoratorManager extends DataProvider
+{
+    public $cache;
+    public $logger;
+
+    /**
+     * @param string $host
+     * @param string $user
+     * @param string $password
+     * @param CacheItemPoolInterface $cache
+     */
+    public function __construct($host, $user, $password, CacheItemPoolInterface $cache)
+    {
+        parent::__construct($host, $user, $password);
+        $this->cache = $cache;
+    }
+
+    public function setLogger(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getResponse(array $input)
+    {
+        try {
+            $cacheKey = $this->getCacheKey($input);
+            $cacheItem = $this->cache->getItem($cacheKey);
+            if ($cacheItem->isHit()) {
+                return $cacheItem->get();
+            }
+
+            $result = parent::get($input);
```

Comment:
Неправильный контекст вызова
```suggestion
            $result = $this->get($input);
```